src/backends/github/API.js

```diff
@@ -64,7 +68,9 @@ export default class API {
     return fetch(url, { ...options, headers }).then((response) => {
       responseStatus = response.status;
       const contentType = response.headers.get("Content-Type");
-      if (contentType && contentType.match(/json/)) {
+      if (url.indexOf('/collaborators/') >= 0) {
+        return responseStatus;
+      } else if (contentType && contentType.match(/json/)) {
```

Comment:
While I understand the motivation behind this code, this is bound to cause bugs in the future by special-casing a particular URL in what is otherwise a generic request function. This should really be done in `collaborator`. One issue I see with the `response` method as it stands is that JSON responses reject the promise when an error status is present, but text responses do not.

@erquhart if you agree that that should be fixed, I can include it in my GH API refactor branch I'm preparing to PR (the branch name is `gh-api-refactors`, there's a little there already).

@tortilaman for now, you should be able to expect a JSON response from the GH API. Thus, any non-200 statuses will cause the promise to reject with the `APIError` in the `catch` block just below these lines. In addition,  My suggestions for refactoring this are as follows:

- Remove this logic from `request`.
- Rename the `collaborator` method to `isCollaborator`, and have that return a promise that either rejects or resolves to a boolean, representing whether the user is a collaborator or not.
- Catch the `404`, `403`, and `204` statuses in `isCollaborator`. If the status is one of those three, resolve the promise to the appropriate boolean. Otherwise, rethrow the error. (remember, JSON responses reject the error if the status is not 200-99, so you don't need any additional 
- Remove the now-unnecessary status code checks from `authenticate` and change them to handle the boolean that `isCollaborator` is returning. Add a `catch` block to handle unexpected errors, and move the `throw new Error('GitHub is not responding, please try again.');` to the `catch` block.

I'd be happy to elaborate on this further if necessary, and I can always refactor this myself if you'd rather move on, but either way I'd like this to be refactored before a merge.